bull/pdl-queue/consumer.js

```diff
@@ -117,22 +126,37 @@ PDLQueue.process(async (job, done) => {
   const trueURI = `http://archive.org/details/${job.data.details.IAIdentifier}`;
   jobLogs["trueURI"] = trueURI;
   jobLogs["userName"] = job.data.details.userName;
-  job.log(JSON.stringify(jobLogs));
-  logUserData(jobLogs["userName"], "Panjab Digital Library");
-  const [zip, byteLength] = await getZipAndBytelength(
+  const [zip, byteLength, errorFlag] = await getZipAndBytelength(
     job.data.details.Pages,
     job.data.details.bookID,
     job.data.details.title,
     job
   );
-  job.progress(90);
-  await uploadToIA(
-    zip,
-    job.data.details,
-    byteLength,
-    job.data.details.email,
-    job
-  );
-  job.progress(100);
-  done(null, true);
+  if (errorFlag.status) {
+    job.log(JSON.stringify(jobLogs));
+    logUserData(jobLogs["userName"], "Panjab Digital Library");
+    logger.log({
+      level: "error",
+      message: `Upload to Internet Archive failed because ${errorFlag.page} is not reachable. Please try again or contact Panjab Digital Library for more details.`,
+    });
+    job.progress(100);
+    done(
+      new Error(
+        `Upload to Internet Archive failed because ${errorFlag.page} is not reachable. Please try again or contact Panjab Digital Library for more details.`
```

Comment:
Can you please add a href to this with target="_blank" so that it opens up in a new tab when user clicks on it?